### ISSUE TITLE
Magento_Search/form-mini not found

### ISSUE THREAD

**Issue by wouter-toppershops** [Smile-SA/elasticsuite#1186]
### Preconditions

Trying to search with autocomplete, gets an requireJs error about file not found.

Magento Version : CE

ElasticSuite Version : 2.6.2

Environment : development

### Steps to reproduce

1. Try search autocompete

### Expected result

1. Get the file

### Actual result

1. 
```
GET http://toppy.docker/static/version1542276727/frontend/Magento/blank/nl_NL/Magento_Search/form-mini.js net::ERR_ABORTED 404 (Not Found)
```

**Comment by wouter-toppershops** [comment]
See Pull https://github.com/Smile-SA/elasticsuite/pull/1187

**Comment by romainruaud** [comment]
Hmmm...

We never faced such an issue.

Can you give us more details about your instance ? Please provide : 

- exact version number of Magento
- exact version number of Elasticsuite
- are you using any custom theme ? if yes, is it based on blank or Luma ? 

Regards

**Comment by wouter-toppershops** [comment]
hi @romainruaud 

we're using Magento 2.3.0-beta32 with Elasticsuite 2.6.x-dev 

we have our own theme. but its also happening on the magento blanc theme.

when i manually change the code line in your module its solves the issue. 

if you look on the magento github, you see they also require it with /js/
https://github.com/magento/magento2/blob/26b744eee7089b3d44a450a2a239a15a75a3e977/app/code/Magento/Search/view/frontend/requirejs-config.js

**Comment by wouter-toppershops** [comment]
and i also see the difference. 

magento moved that file somewhere in 2.2.x
https://github.com/magento/magento2/blob/2.2/app/code/Magento/Search/view/frontend/requirejs-config.js

**Comment by wouter-toppershops** [comment]
https://github.com/magento/magento2/commit/8653ba1fbe29acc37505b85adce1463d70e9cdf1#diff-df5888442619e22b6f135ff4e932a86e

**Comment by wouter-toppershops** [comment]
See: https://github.com/magento/magento2/pull/16582

**Comment by romainruaud** [comment]
Ok... so it seems to be due to Magento having moved this file... Nice for compatibility backwards ...

We'll fix this one as part of our 2.3.0 compatibility.

Thank you for pointing this to us.

**Comment by romainruaud** [comment]
@rbayet can you have a look on this one ? 

The PR #1187 obviously fixes it on latest Magento version (2.2.6 and 2.3.0 beta) but I'd like to have something more smooth and of course something able to deal with backward compatibility.

Not sure it's possible actually ..... Maybe we can rely on something hacky for now, and fix it definitely later when we'll decide to drop support of Magento versions < 2.2.6.

**Comment by rbayet** [comment]
Actually, I don't see the change being done in the tag 2.2.6 yet, but the changes have definitely been made in the 2.2-develop branch.
Following the BC approach described in https://github.com/magento/magento2/pull/16582 should do the trick.

**Comment by rbayet** [comment]
@romainruaud I focused on a smooth M2 2.2.x approach in PR #1190 but it will *not* work on 2.3 where the "old : new " path mapping is absent in Magento_Search.
I don't see a way to have a common hack working for both 2.2.x and 2.3.x : introducing ourselves the path mapping would work in 2.3x, probably do nothing in versions >= 2.2.7, but break in 2.2.6 and before.

So it looks like a new branch is required just for that fix and the other upcoming 2.3 fixes to come...

**Comment by wouter-toppershops** [comment]
@rbayet could you please make that branch with only this fix for now?
magento will release 2.3 very soon. this year according to the planning.

**Comment by romainruaud** [comment]
Hello,

so the issue is actually solved for 2.2 version of Magento, and BC compliant for all 2.2.x

If we do not find an hybrid or hacky way to get it working on Magento 2.3.0, I'm okay to have a different fix of this one being implemented into master branch (future 2.7.0 major), knowing that it would leverage our requirements to Magento > 2.3.0.

In the meantime, we will also contact Magento to ensure they did not report the "old : new" path mapping in 2.3.0, therefore creating a BC issue for us, but also for many people building extensions on top of their solution. Not sure they'll be able to provide a fix on time, or even to release another beta, since it's expected to be live on monday ...

**Comment by rbayet** [comment]
@wiardvanrij 2.6.4 is going to be released very soon, probably next Thursday (CET).
